### PR TITLE
Move `integration/allennlp`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Move `integration/allennlp` from optuna/optuna

## Description of the changes
Move `integration/allennlp` from optuna/optuna

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,12 @@ integrated with many useful tools like PyTorch, sklearn, TensorFlow, etc.
 
 Optuna-Integration API reference is [here](https://optuna.readthedocs.io/en/stable/reference/integration.html).
 
+* [AllenNLP](https://optuna.readthedocs.io/en/stable/reference/integration.html#allennlp)  ([example](https://github.com/optuna/optuna-examples/tree/main/allennlp))
 * [Chainer](https://optuna.readthedocs.io/en/stable/reference/integration.html#chainer)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainer_integration.py))
 * [ChainerMN](https://optuna.readthedocs.io/en/stable/reference/integration.html#chainermn)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainermn_simple.py))
 * [Keras](https://optuna.readthedocs.io/en/stable/reference/integration.html#keras)  ([example](https://github.com/optuna/optuna-examples/tree/main/keras))
 
+
```
Comment (gen740) [inline README.md:23]: This is a bit trivial, but please remove this unnecessary new line.

File: docs/source/reference/index.rst [hunk 2]
```diff
@@ -23,6 +23,9 @@ Chainer
    :toctree: generated/
    :nosignatures:
 
+   optuna.integration.AllenNLPExecutor
+   optuna.integration.allennlp.dump_best_config
+   optuna.integration.AllenNLPPruningCallback
```
Comment (HideakiImamura) [inline docs/source/reference/index.rst:28]: These modules are listed inside the `Chainer` section. I think we should create the `AllenNLP` section.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. Almost, LGTM. Please check one minor comment.

**Review by gen740 — COMMENTED** [review]
Almost LGTM! But I have a minor comment.

**Comment by Alnusjaponica** [comment]
Thank you for your review. I updated the changes accordingly.

**Review by gen740 — APPROVED** [review]
LGTM!

**Pull request merged**